schema/workflow.json

```diff
@@ -442,17 +442,14 @@
           "description": "Reference to the unique name of a 'consumed' event definition"
         },
         "data": {
-          "type": [
-            "string",
-            "object"
-          ],
-          "description": "If string type, an expression which selects parts of the states data output to become the data (payload) of the event referenced by 'triggerEventRef'. If object type, a custom object to become the data (payload) of the event referenced by 'triggerEventRef'."
+          "$ref": "common.json#/definitions/objectOrMarkup",
```

Comment:
why does it have to be object or markup restriction? sorry i don't understand

Reply:
I think we want to allow expressions or objects. "objectOrMarkup" is pretty much that, it allows to either specify an object or an expression. But also, it allows to specify objects that is built from several expressions:
```json
"data": {
  "foo": "bar",
  "dynamic": "${ .someValue }"
}
```